src/nightwatch/nightwatch.config.js

```diff
@@ -1,27 +1,46 @@
 /* eslint import/no-extraneous-dependencies: ["error", {"devDependencies": true}] */
 /* eslint no-unused-vars: ["error", { "varsIgnorePattern": "chromedriver" }] */
 
-import chromedriver from 'chromedriver';
 import webpack from 'webpack';
 import WebpackDevServer from 'webpack-dev-server';
+import ip from 'ip';
+
+import SeleniumDockerService from '../wdio/services/SeleniumDockerService';
 
 let port = 8080;
 
+const seleniumPort = 4444;
+const seleniumHost = ip.address();
+
 const nightwatchConfig = (webpackConfig, srcFolders, providedPort) => {
   if (providedPort) {
     port = providedPort;
   }
 
+  const seleniumDocker = new SeleniumDockerService();
   const webpackServer = new WebpackDevServer(webpack(webpackConfig), { quiet: true, hot: false, inline: false });
 
   const startDriverAndServer = (done) => {
-    chromedriver.start();
-    webpackServer.listen(port, '0.0.0.0', () => done());
+    const staticServerPromise = new Promise((resolve) => {
+      webpackServer.listen(port, '0.0.0.0', () => resolve());
+    });
+
+    const dockerPromise = seleniumDocker.onPrepare({
+      host: seleniumHost,
+      port: seleniumPort,
+      path: '/wd/hub',
+      seleniumDocker: {
+        enabled: !process.env.TRAVIS && !process.env.CI,
+        cleanup: true,
+      },
+    }, [{ browserName: 'chrome' }]);
+
+    Promise.all([staticServerPromise, dockerPromise]).then(() => done());
```

Comment:
I think you could just pass in done to then.  What happens if either of the promises error out?